uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-balance-tab.component.js

```diff
@@ -144,6 +151,18 @@ define([
     }
 
     /**
+     * Sets the user's role based on their permissions
+     *
+     * @return {Promise}
+     */
+    function setUserRole () {
```

Comment:
why not add this logic to the filters component? The filter component should pass all parameters necessary to filter the request. This component should worry about retrieving and displaying the records only.

Reply:
Remember that the filter component is not just a for visual filters.

Reply:
> Remember that the filter component is not just a for visual filters.

I am not quite sure it is. It produces filters and then passes it to the master component. It is the master component that calls API via `loadReportCurrentPage()` basing on these filters. 
That's actually why I decided to delete `managed_by` from `filters = {}` since I realised it is not a filter, but a property that is needed to call the API.